Sources/ZippyJSON/ZippyJSONDecoder.swift

```diff
@@ -1057,3 +1078,63 @@ extension __JSONDecoder : SingleValueDecodingContainer {
 
     // End
 }
+
+@propertyWrapper
+private final class Synchronized<Value>: @unchecked Sendable {
+    private let lock: LockProtocol
+    private var _wrappedValue: Value
+
+    init(wrappedValue: Value) {
+        self.lock = Lock()
+        _wrappedValue = wrappedValue
+    }
+
+    var wrappedValue: Value {
+        get {
+            lock.withLock {
+                _wrappedValue
+            }
+        }
+        set {
+            lock.withLock {
+                _wrappedValue = newValue
+            }
+        }
+    }
+}
+
+private final class Lock: LockProtocol {
```

Comment:
i think we could remove this class and just have a free function that returns a LockProtocol